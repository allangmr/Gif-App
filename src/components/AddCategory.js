import React, { useState } from 'react'
import Proptypes from 'prop-types'


function AddCategory({setCategories}) {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = e =>{
        setInputValue(e.target.value)
    }
    const handleSubmit = (e) =>{
        e.preventDefault();
        if( inputValue && inputValue.trim().length >2){
            setCategories(prev => [inputValue, ...prev])
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: Proptypes.func.isRequired
}

export default AddCategory
