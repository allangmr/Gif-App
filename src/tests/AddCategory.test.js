import { shallow } from "enzyme";
import AddCategory from "../components/AddCategory";

describe('Test in the AddCategory.js', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })
    test('should be show correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change of input', () => {
        const value = 'Hola Mundo';
        wrapper.find('input').simulate('change', {target: { value }});
        expect(wrapper.find('p').text().trim()).toBe('Hola Mundo');

    });

    test('should not send the information on submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault: () => {}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should be call the setCategories and clean the input text', () => {
        // 1. Simular el inputChange
        const value = 'Hola Mundo';
        wrapper.find('input').simulate('change', {target: { value }});
        expect(wrapper.find('p').text().trim()).toBe('Hola Mundo');
        // 2. Simular el submit
        wrapper.find('form').simulate('submit', {preventDefault: () => {}});
        // 3. setCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        //4. El valor del input debe estar en ''
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});