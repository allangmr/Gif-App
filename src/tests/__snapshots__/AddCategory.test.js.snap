// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the AddCategory.js should be show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
